File: problems/002/answer.r
#!/usr/bin/env Rscript

##
# Author: Caleb Adams
# Discription: This is a very simple algorithm that runs in
# linear time O(n) to compute the sum of all even fib numebrs
## 

ret <- 0
fib_a <- 0
fib_b <- 1
fib_c <- 1
max <- 4000000
while (fib_c < max){
      fib_c <- fib_a + fib_b
      fib_a <- fib_b
      fib_b <- fib_c
      if (fib_c %% 2 == 0){
      	 ret <- ret + fib_c
      }
}

cat("answer: ", ret)
cat("\n")

File: problems/001/answer.java
//package 001;

public class answer {
    public static void main(String[] args) {
	int ret = 0;
	for (int i = 0; i < 1000; i++){
	    if (i % 3 == 0 || i % 5 == 0){
		ret += i;
	    }
	}

	System.out.println("answer: " + ret);	
    }
}
